.github/workflows/magma-build-openvswitch.yml

```diff
@@ -0,0 +1,110 @@
+# Copyright 2022 The Magma Authors.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Magma Build & Publish Open vSwitch Artifacts
+on:
+  push:
+    branches:
+      - master
+      - 'v1.*'
+    paths:
+      - third_party/gtp_ovs/ovs-gtp-patches/**
+      - .github/workflows/magma-build-openvswitch.yml
+  pull_request:
+    types: [ opened, reopened, synchronize ]
+    branches:
+      - master
+      - 'v1.*'
+    paths:
+      - third_party/gtp_ovs/ovs-gtp-patches/**
+      - .github/workflows/magma-build-openvswitch.yml
+  schedule:
+    - cron: 0 4 * * 0
+  workflow_dispatch:
+    inputs:
+      production:
+        description: Is production build?
+        type: boolean
+        default: false
+      repository:
+        description: Repository to push to?
+        type: choice
+        default: magma-packages-test
+        options: [ magma-packages-test, magma-packages-prod ]
+        required: true
+      distribution:
+        description: Distribution to set?
+        default: 'focal-ci'
+        required: true
+
+jobs:
+  build_packages:
+    runs-on: ubuntu-20.04
+    env:
+      MAGMA_ROOT: "${{ github.workspace }}"
+      DEST: "${{ github.workspace }}/deb_packages"
+      repository: ${{ inputs.repository || 'magma-packages-test' }}
+      distribution: ${{ inputs.distribution || 'focal-ci' }}
+    steps:
+      - uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # pin@v3.1.0
+
+      - name: Build openvswitch
+        working-directory: third_party/gtp_ovs/ovs-gtp-patches/2.15
+        run: |
+          mkdir ${{ env.DEST }}
+          ./build.sh ${{ env.DEST }}
+
+      - name: Test installability of the debian package
+        working-directory: ${{ env.DEST }}
+        run: |
+            sudo pip install ovs==2.16.0
+            sudo apt-get install --yes --allow-downgrades ./*.deb
+
+      - name: Set repository to debian-test if not workflow_dispatch
+        run: echo "repository=debian-test" >> $GITHUB_ENV
+        if: github.ref == 'refs/heads/master' && github.event_name != 'workflow_dispatch'
```

Comment:
Do we need to check if the repository owner is magma?

Reply:
Maybe this is not needed since here we set the repository which is only be used to upload the artifacts but we need to be in the magma repo to get access to the LF artifactory secrets.